### DISCUSSION TITLE
Extreme low precisinon of GPS on GoPro 9 Hero Black

### DISCUSSION THREAD

**Discussion by tommyhustava** [gopro/labs#787] [General]
Hi, i have loaded firmware 1.72.xx on my Hero 9 Black camera. I have problem with precision in GPS data. On pictures, really footage is from Hronska Dubrava, Slovakia. With GPS coordinastes, tne location is in Poland, cca 300 km from me. It is possible?
![GX040160 MP4_snapshot_06 16 958](https://github.com/gopro/labs/assets/116656429/4a086647-f5dc-4571-9090-45d42c598add)
![hronska dobrava](https://github.com/gopro/labs/assets/116656429/d464506d-a39a-4d55-8018-8594d6f62994)

**Comment by dnewman-gpsw** [comment]
This is not a Labs issue, unless you are saying Labs lowered the precision. GPS takes some time to lock, camera needs to be on for about a minute, outside, camera orientation shutter button up, and not recording.  Don't start with QuickCapture as that can delay a good GPS lock, forever.  This did get better on H10 and H11.  HERO9 was way slower to lock while recording.  There is an attempted Labs improvement for HERO9, try !MGPSF=1

**Comment by tommyhustava** [comment]
Well thank you. I understand... But I'm very used to the top button, I
always start and end recording with this button. What does the "!MGPSF=1"
parameter actually do?

Dňa ut 27. 2. 2024, 17:56 David Newman ***@***.***>
napísal(a):

**Reply by dnewman-gpsw** [reply]
It is complicated. It attempts to improve GPS reception by changing the clock frequencies the camera is using.  It is only a HERO9 feature.

**Comment by tommyhustava** [comment]
I'll try it and let you know.

Dňa ut 27. 2. 2024, 18:36 David Newman ***@***.***>
napísal(a):

**Comment by tommyhustava** [comment]
Hi.

The command " !MGPSF=1 " are accepted by device, but the precision of GPS
is same - very bad and without any change, with diameter plus minus 300
kilometers from my real position.

I entered the parameter into the device via QR code separately. Did I do it
right? Or does it need to be entered in the string I use to display the
overlay?
My string is: g1!MLFIN=0!MBRNO=10!MBURN="(0,22)TOMMY [BRdd.mm.yyyy HH:MM:SS
| GPS N'GPS50%1.9f' E'GPS51%1.9f' S'GPS53%1.2f']"

Every day I travel 60km to work by bicycle between the cities of Žiar nad
Hronom and Zvolen, (Slovakia) and neither in the morning nor in the
afternoon do I have a few minutes for the camera to capture the correct
GPS. I just simply turn on the recording with the top button and go. GPS is
therefore unusable for me, which is a shame...

ut 27. 2. 2024 o 18:36 David Newman ***@***.***> napísal(a):

**Reply by dnewman-gpsw** [reply]
I run some tests on HERO9 and HERO11, both had correct metadata in the MP4 track, but only HERO11 rendered the overlay correctly.   This appears to be Labs bug on HERO9.  

You might try this instead: https://goprotelemetryextractor.com/telemetry-overlay-trial

**Comment by tommyhustava** [comment]
Thank you for addressing this issue in such detail. If this is a problem in
Labs for the Hero9, is there any chance that this bug in rendering GPS data
to the overlay will be fixed? I'll tell you the truth, I didn't think to
check the GPS metadata from the video file with the ones rendered into the
overlay.

Dňa pi 1. 3. 2024, 20:04 David Newman ***@***.***> napísal(a):

**Reply by dnewman-gpsw** [reply]
There is very little chance any GP1 cameras will have further Labs updates.  Labs is generally updated only when the main firmware is updated.

**Comment by tommyhustava** [comment]
I understand. The Hero9 is a great camera and Labs gives it great options.
It's a shame they'll probably never fix it.

Dňa pi 1. 3. 2024, 21:21 David Newman ***@***.***> napísal(a):

**Comment by nanophilip** [comment]
@tommyhustava, are the wrong GPS coordinates shown on the overlay of the GoPro's display, or are the GPS coordinates embedded in the video file wrong? If the former, you might want to try an older Labs firmware. That fixed a similar (same?) [issue](https://github.com/gopro/labs/issues/1677) for me.

**Comment by tommyhustava** [comment]
thank you!

st 27. 5. 2026 o 5:49 Philip Düe ***@***.***> napísal(a):
